modules/osdk-control-compat.adoc

```diff
@@ -0,0 +1,88 @@
+
+[IMPORTANT]
+====
+You should configure the following settings in your Operator project as soon as possible. Users running workloads with deprecated APIs that are planning to eventually upgrade to a future version of {product-title} must be running an Operator that has the proper configuration to avoid a cluster upgrade to an incompatible version and potentially adversely impact their critical workloads.
+====
+
+.Prerequisites
+
+- Operator SDK CLI installed on a development workstation
+- Operator project initialized by using the Operator SDK
+
+.Procedure
+
+. Configure the maximum version of {product-title} that your Operator is compatible with. In your Operator project's cluster service version (CSV), set the `olm.openShiftMaxVersion` annotation to prevent administrators from upgrading their cluster before upgrading the installed Operator to a compatible version:
++
+.Example CSV with `olm.maxOpenShiftVersion` annotation
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    "olm.properties": '[{"type": "olm.maxOpenShiftVersion", "value": "4.8"}]' <1>
+----
+<1> Setting `value` to `4.8` prevents cluster upgrades to {product-title} versions later than 4.8 when this bundle is installed on a cluster.
+
+. If your bundle is intended for distribution in a Red Hat-provided Operator catalog, configure the compatible versions of {product-title} for your Operator by setting the following properties. This configuration ensures your Operator is only included in catalogs that target compatible versions of {product-title}:
++
+[NOTE]
+====
+This step is only valid when publishing Operators in Red Hat-provided catalogs. If your bundle is only intended for distribution in a custom catalog, you can skip this step. For more details, see "Red Hat-provided Operator catalogs".
+====
+
+.. Set the `com.redhat.openshift.versions` annotation in your project's `bundle/metadata/annotations.yaml` file:
++
+.Example `bundle/metadata/annotations.yaml` file with compatible versions
+[source,yaml]
+----
+com.redhat.openshift.versions: "v4.6-v4.8" <1>
+----
+<1> Set to a range or single version.
+
+.. To prevent your bundle from being carried on to an incompatible version of {product-title}, ensure that the index image is generated with the proper `com.redhat.openshift.versions` label in your project's `bundle.Dockerfile`:
++
+.Example `bundle.Dockerfile` with compatible versions
++
+[source,yaml]
+----
+LABEL com.redhat.openshift.versions="v4.6-v4.8" <1>
+----
+<1> Set to a range or single version.
++
+[NOTE]
+====
+This label is also useful when you know that the current version of your Operator will not work well, for any reason, on a specific {product-title} version. By using it, you define the cluster versions where the Operator should be distributed, and the Operator does not appear in a catalog of a cluster version which is outside of the range.
```

Comment:
The text `This label is also useful when you know that the current version of your Operator will not work well` actually would better fit in a not for the max OCP version the idea of that is to allow the operator author to define the max OCP version supported by their specific versions of the distribution.

Reply:
Took this out of the "NOTE" format and moved it up into the proper intro paragraphs right before the beginning of the procedure. See https://github.com/openshift/openshift-docs/pull/34348/files#diff-9a5395e4fb38f80963c37a932625ff0a705ed827f3e2ecc59296e085d709ab44R32.